code/modules/atmos_automation/implementation/vent_pump.dm

```diff
@@ -1,95 +1,126 @@
 
 /datum/automation/set_vent_pump_mode/Topic(href,href_list)
+	. = ..()
+	if(.)
+		return
+
 	if(href_list["set_mode"])
 		mode = input("Select a mode to put this pump into.",mode) in modes
 		parent.updateUsrDialog()
 		return 1
+
 	if(href_list["set_vent_pump"])
-		var/list/injector_names=list()
-		for(var/obj/machinery/atmospherics/unary/vent_pump/I in atmos_machines)
-			if(!isnull(I.id_tag) && I.frequency == parent.frequency)
-				injector_names|=I.id_tag
-		for(var/obj/machinery/atmospherics/binary/dp_vent_pump/I in atmos_machines)
-			if(!isnull(I.id_tag) && I.frequency == parent.frequency)
-				injector_names|=I.id_tag
+		var/list/injector_names = list()
+
+		if(!mode)
+			for(var/obj/machinery/atmospherics/unary/vent_pump/I in atmos_machines)
+				if(!isnull(I.id_tag) && I.frequency == parent.frequency)
+					injector_names |=I.id_tag
+
+		else
+			for(var/obj/machinery/atmospherics/binary/dp_vent_pump/I in atmos_machines)
+				if(!isnull(I.id_tag) && I.frequency == parent.frequency)
+					injector_names |= I.id_tag
+
 		vent_pump = input("Select a vent:", "Vent Pumps", vent_pump) as null|anything in injector_names
 		parent.updateUsrDialog()
 		return 1
 
+	if(href_list["toggle_type"])
+		vent_type = !vent_type
+		parent.updateUsrDialog()
+		return 1
+
 /datum/automation/set_vent_pump_power
 	name="Vent Pump: Power"
 
-	var/vent_pump=null
-	var/state=0
+	var/vent_pump = null
+	var/state = 0
+	var/mode = 0//0 for unary vents, 1 for DP vents.
 
 /datum/automation/set_vent_pump_power/Export()
 	var/list/json = ..()
-	json["vent_pump"]=vent_pump
-	json["state"]=state
+	json["vent_pump"] = vent_pump
+	json["state"] = state
+	json["mode"] = mode
 	return json
 
 /datum/automation/set_vent_pump_power/Import(var/list/json)
 	..(json)
 	vent_pump = json["vent_pump"]
 	state = text2num(json["state"])
-
-/datum/automation/set_vent_pump_power/New(var/obj/machinery/computer/general_air_control/atmos_automation/aa)
-	..(aa)
+	mode = text2num(json["mode"])
 
 /datum/automation/set_vent_pump_power/process()
 	if(vent_pump)
-		parent.send_signal(list ("tag" = vent_pump, "power"=state), RADIO_FROM_AIRALARM)
+		parent.send_signal(list ("tag" = vent_pump, "power" = state, "type" = "vent"), filter = (mode ? RADIO_ATMOSIA : RADIO_FROM_AIRALARM))
 
 /datum/automation/set_vent_pump_power/GetText()
-	return "Set vent pump <a href=\"?src=\ref[src];set_vent_pump=1\">[fmtString(vent_pump)]</a> power to <a href=\"?src=\ref[src];set_power=1\">[state ? "on" : "off"]</a>."
+	return "Set <a href=\"?src=\ref[src];toggle_mode=1\">[mode ? "Dual-Port" : "Unary"]</a> vent pump <a href=\"?src=\ref[src];set_vent_pump=1\">[fmtString(vent_pump)]</a> power to <a href=\"?src=\ref[src];set_power=1\">[state ? "on" : "off"]</a>."
 
 /datum/automation/set_vent_pump_power/Topic(href,href_list)
+	. = ..()
+	if(.)
+		return
```

Comment:
I believe that only works if you don't specify a return otherwise.
In this case, you're specifying a null return.